Documentation/WorkflowSwiftUI Adoption Guide.md

```diff
@@ -0,0 +1,448 @@
+# WorkflowSwiftUI Adoption Guide
+
+`WorkflowSwiftUI` is designed with performance in mind. Instead of updating the entire UI on every render loop, `WorkflowSwiftUI` uses the [Perception](https://github.com/pointfreeco/swift-perception) framework (a backport of Apple's Observation), to detect the data that each view is dependent on, and to only re-evaluate those views when that data has changed.
+
+In order to support this, `WorkflowSwiftUI` has some additional restrictions and subtle departures from practices you may be used to.
+
+- Your *view* and *workflow* both have knowledge of *state* and *actions* (collectively, a “model”).
+- Your workflow renders a *model* rather than a *screen*.
+- We recommend defining your model, view, and screen independently, rather than nesting any of these types.
+
+## State
+
+Observation is used to detect when the data that views use has changed, and to re-evaluate only the views that are affected.
+
+To achieve this, your workflow’s `State` type must be annotated with the `@ObservableState` macro. This macro adds observation hooks to each property, and also endows your state struct with a concept of identity. 
+
+```swift
+@ObservableState
+public struct State {
+  var foo: String
+  var bar: Int
+}
+```
+
+For an `ObservableState` type, creating a copy does not change the state’s identity in the copy. However, creating a new value by calling the initializer will result in a new identity.
+
+If your state has nested types, you can annotate them with `@ObservableState` as well, for fine-grained observation of the properties of that type as well.
+
+```swift
+@ObservableState
+public struct State {
```

Comment:
Remind me, but does Apple's observation stuff support `structs` yet, or is it just classes?

Reply:
Yep, just classes, that's why we need our own `ObservableState` macro.